plugins/crowdstrike_falcon_intelligence/icon_crowdstrike_falcon_intelligence/util/api.py

```diff
@@ -0,0 +1,164 @@
+        return self.make_json_request(
+            method="GET",
+            url=self._base_url + REPORTS_QUERY_ENDPOINT,
+            params={"limit": limit, "offset": offset, "filter": filter_query},
+        ).get("resources")
+
+    def get_submissions_ids(self, offset: int = None, limit: int = None, filter_query: str = None) -> list:
+        self._logger.info("Getting a list of submission IDs...")
+        return self.make_json_request(
+            method="GET",
+            url=self._base_url + SUBMISSIONS_QUERY_ENDPOINT,
+            params={"limit": limit, "offset": offset, "filter": filter_query},
+        ).get("resources")
+
+    def make_request(
+        self,
+        method: str,
+        url: str,
+        params: dict = None,
+        json_data: dict = None,
+    ) -> requests.Response:
+        try:
+            response = requests.request(
+                method=method,
+                url=url,
+                headers=self.get_headers(),
+                params=clean_dict(params),
+                json=clean_dict(json_data),
+            )
+
+            if response.status_code == 400:
+                raise PluginException(preset=PluginException.Preset.BAD_REQUEST, data=response.text)
+            if response.status_code == 403:
+                raise PluginException(
+                    cause="Operation is not allowed.",
+                    assistance="Please verify inputs and if the issue persists, contact support.",
+                    data=response.text,
+                )
+            if response.status_code == 404:
+                raise PluginException(
+                    cause="Resource not found.",
+                    assistance="Please verify inputs and if the issue persists, contact support.",
+                    data=response.text,
+                )
+            if 400 <= response.status_code < 500:
+                raise PluginException(
+                    preset=PluginException.Preset.UNKNOWN,
+                    data=response.text,
+                )
+            if response.status_code >= 500:
+                raise PluginException(preset=PluginException.Preset.SERVER_ERROR, data=response.text)
+
+            if 200 <= response.status_code < 300:
+                return response
+
+            raise PluginException(preset=PluginException.Preset.UNKNOWN, data=response.text)
+        except requests.exceptions.HTTPError as e:
+            raise PluginException(preset=PluginException.Preset.UNKNOWN, data=e)
+
+    def make_json_request(
+        self, method: str, url: str, params: dict = None, json_data: dict = None
+    ) -> Union[list, dict]:
+        try:
+            response = self.make_request(method=method, url=url, params=params, json_data=json_data)
+            return response.json()
+        except json.decoder.JSONDecodeError as e:
+            raise PluginException(preset=PluginException.Preset.INVALID_JSON, data=e)
```

Comment:
Just to stay with one concept. Kindly please do not use single variable names 😄 
```suggestion
        except json.decoder.JSONDecodeError as error:
            raise PluginException(preset=PluginException.Preset.INVALID_JSON, data=error)
```